crates/nu-test-support/src/macros.rs

```diff
@@ -239,7 +238,10 @@ pub fn nu_run_test(opts: NuOpts, commands: impl AsRef<str>, with_std: bool) -> O
     let mut paths = crate::shell_os_paths();
     paths.insert(0, test_bins);
 
-    let commands = commands.as_ref().lines().collect::<Vec<_>>().join("; ");
+    let commands = match opts.join {
+        None | Some(true) => commands.as_ref().lines().collect::<Vec<_>>().join("; "),
+        Some(false) => commands.as_ref().into(),
+    };
```

Comment:
Do I understand this correctly is this a workaround for what we have with `pipeline()` already?

Reply:
I'm not sure what you're referring to, but this is here because the command given to the `nu!` macro is joined with `;`s which makes multi line commands impossible. I thought it best to include this change in the PR because otherwise, the test tables have to be all on a single line, which makes it very difficult to read.

Though I am happy to split that change off into a separate PR and base this on top.

Reply:
For multiline pipelines we have the weird helper function [`nu_test_support::pipeline`](https://docs.rs/nu-test-support/latest/nu_test_support/fn.pipeline.html) that will take the pipeline and turn it into a single line so `nu!` can safely pass it to `nu -c`. Haven't checked yet if this would work for all your new tests.

Reply:
Ah, I hadn't seen that helper function. I think for these tests, that would work, since it's going to join them all into one line before passing to `nu!`. I'm not really sure I understand why it's necessary to make the input script all on a single line, except maybe as a workaround to this same issue of the semicolons being injected and breaking the input script. The command passed into `-c` need not be on a single line to work.

Personally, I still think it's beneficial to provide this knob to opt out of massaging the test script. When I write nu code in a test, I expect the code to be run as I wrote it. I'm not sure what the original use case was for doing this, but changing the input script like this on the face of it seems to me like asking for, at best, confusion, or worse, creating silently unsound tests by changing its semantics in a way that the programmer was not aware of.